lib/flight_job/commands/edit_script.rb

```diff
@@ -0,0 +1,76 @@
+#==============================================================================
+# Copyright (C) 2021-present Alces Flight Ltd.
+#
+# This file is part of Flight Job.
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which is available at
+# <https://www.eclipse.org/legal/epl-2.0>, or alternative license
+# terms made available by Alces Flight Ltd - please direct inquiries
+# about licensing to licensing@alces-flight.com.
+#
+# Flight Job is distributed in the hope that it will be useful, but
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, EITHER EXPRESS OR
+# IMPLIED INCLUDING, WITHOUT LIMITATION, ANY WARRANTIES OR CONDITIONS
+# OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY OR FITNESS FOR A
+# PARTICULAR PURPOSE. See the Eclipse Public License 2.0 for more
+# details.
+#
+# You should have received a copy of the Eclipse Public License 2.0
+# along with Flight Job. If not, see:
+#
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on Flight Job, please visit:
+# https://github.com/openflighthpc/flight-job
+#==============================================================================
+
+require 'tty-editor'
+
+module FlightJob
+  class Commands::EditScript < Command
+    def run
+      # Ensure the script exists up front
+      script
+
+      # Opens vim/emacs/nano commands at the start line, else use the regular editor
+      # NOTE: gedit needs x-forwarding over ssh, but it also supports the +line
+      cmd = if ['vi', 'vim', 'nvim', 'emacs', 'nano', 'gedit'].include?(editor)
+              if start_line
+                "#{editor} +#{start_line}"
+              else
+                editor
+              end
+            else
+              editor
+            end
+
+      # Open the file
+      TTY::Editor.open(script.script_path, command: cmd)
+    end
+
+    def script
+      @script ||= load_script(args.first)
+    end
+
+    # Determine which line to open the script on
+    def start_line
+      File.open(script.script_path) do |file|
+        _, idx = file.each_line.each_with_index.find do |line, _|
+          />{4,}.*<{4,}/.match?(line)
```

Comment:
This pattern match doesn't work for any scripts I created in prior versions of flight-job.  Matching against `/^# *>{4,}.*WORKLOAD` should be robust against most changes to templates and still work with the existing scripts.